src/pf-modal/pf-modal.spec.js

```diff
@@ -0,0 +1,139 @@
+        spyOn(customElement, '_hideModal');
+        customElement.setAttribute('open', '');
+        customElement.open = false;
+        expect(customElement._hideModal).toHaveBeenCalled();
+      });
+    });
+
+    it('Reflecting attributes to properties', function () {
+      customElement.setAttribute('open', '');
+      expect(customElement.open).toBe(true);
+
+      customElement.removeAttribute('open');
+      expect(customElement.open).toBe(false);
+
+      customElement.removeAttribute('backdrop');
+      expect(customElement.backdrop).toBe(false);
+
+      customElement.setAttribute('backdrop', '');
+      expect(customElement.backdrop).toBe(true);
+
+      customElement.removeAttribute('keyboard');
+      expect(customElement.keyboard).toBe(false);
+
+      customElement.setAttribute('keyboard', '');
+      expect(customElement.keyboard).toBe(true);
+    });
+  });
+
+  describe('Methods testing', function () {
+
+    it('modal should be visible after calling method show()', function() {
+      customElement.show();
+      expect(customElement.getAttribute('open')).toBe('');
+    });
+
+    describe('Method hide() testing', function () {
+      it('_hideModal() should be called when hide() is called', function () {
+        spyOn(customElement, '_hideModal');
+        customElement.open = true;
+        customElement.hide();
+        expect(customElement._hideModal).toHaveBeenCalled();
+      });
+      it('Aattribute open will be removed after calling method _afterHideModal', function () {
+        customElement._afterHideModal();
+        expect(customElement.getAttribute('open')).toBe(null);
+      });
+    });
+
+    it('show() method should be called when calling method toggle() on visible modal', function () {
+      spyOn(customElement, 'show');
+      customElement.toggle();
+      expect(customElement.show).toHaveBeenCalled();
+    });
+  });
+
+  describe('Events testing', function () {
+    var callback = null;
+
+    beforeEach(function () {
+      callback = jasmine.createSpy();
+    });
+
+    it('show.bs.modal should be triggered', function () {
+      customElement.addEventListener('show.bs.modal', callback);
+      customElement.show();
+      expect(callback).toHaveBeenCalled();
+      expect(callback.calls.mostRecent().args[0].detail).toBeNull();
+    });
+
+    it('shown.bs.modal should be triggered', function () {
+      customElement.addEventListener('shown.bs.modal', callback);
+      customElement._afterShowModal();
+      expect(callback).toHaveBeenCalled();
+    });
+
+    it('hide.bs.modal should be triggered', function () {
+      customElement.addEventListener('hide.bs.modal', callback);
+      customElement.hide();
+      expect(callback).toHaveBeenCalled();
+    });
+
+    it('hidden.bs.modal should be triggered', function () {
+      customElement.addEventListener('hidden.bs.modal', callback);
+      customElement._afterHideModal();
+      expect(callback).toHaveBeenCalled();
+    });
+  });
+
```

Comment:
Now that we're providing default cancel, save, and close buttons, we should test the events here. For example, we can test if clicking the close button hides the modal. Users should be able to receive an event for save, cancel, etc.